### ISSUE TITLE
Source code for DIPS split

### ISSUE THREAD

**Issue by anton-bushuiev** [octavian-ganea/equidock_public#18]
Hi 👋! In the paper it is mentioned: "For DIPS, the split is based on protein family to separate similar proteins". Is there a source code for this split? I could only find a [random split in `paritition_dips.py`](https://github.com/octavian-ganea/equidock_public/blob/main/src/utils/partition_dips.py#:~:text=random.shuffle,%2D20%3A%5D).

**Comment by AxelGiottonini** [comment]
Hey !

I don't remember finding any code for the split, but you can certainly use create a simple script to cluster your proteins using foldseek or something similar and dgl, networkx or any other graph library you want. The only thing you need to output is then the list of files in the same format than you could find in the original splits definition.

Sincerly meow !

**Comment by anton-bushuiev** [comment]
Hi, @AxelGiottonini!

Thank you very much for you response. Foldseek looks perfect, I did not know about it. What exactly do you mean by using a graph library? To cluster PPIs using graph metrics based on their EquiDock graph representations? Also, I am still curios how exactly PPIs were split based on the folds of individual interacting partners. If `PPI1` has partners with folds `A` and `B` and `PPI2` with `C` and `D`, are they decided to be separated if {`A`, `B`} != {`C`, `D`} or more strictly {`A`, `B`} and {`C`, `D`} are disjoint 🤔? It may be important from the perspective of data leakage.

**Comment by AxelGiottonini** [comment]
What I did in a previous project was to cluster the proteins using foldseek (all vs all) and to create a graph using all the protein as  vertices and putting edges between paired proteins (receptor - ligand) and proteins in a cluster. Then I used the biggest clusters to create the training set and the smallest for validation and testing (90-5-5). 

What may be an option could also be to characterize the binding pocket and split the data according to this characterization, but I miss knowledge to do that kind of things.

**Comment by anton-bushuiev** [comment]
Thank you for sharing!

Yes, I am also considering to create a split based on interface similarity using a tool like [this](https://github.com/haddocking/arctic3d).

**Comment by AxelGiottonini** [comment]
You're welcome ! I did not look for such tool but that seems promising !

Also, when I was working with EquiDock, I had results with a bad accuracy considering only the ligand RMSD (as the receptor RMSD is always 0). I'll share my code and results in the next days, but could you consider sharing your results if something similar occurred?

**Comment by anton-bushuiev** [comment]
Hi! I do not use EquiDock and I was mainly interested in the data split. I am working on a related problem of predicting binding affinity change upon mutation (based on the [SKEMPI2](https://life.bsc.es/pid/skempi2) data). It as about learning from already bound structures, so its a bit different.
